ui/js/controllers/perf/compare.js

```diff
@@ -306,12 +306,12 @@ perf.controller('CompareResultsCtrl', [
                         subtests: 0,
                         framework: $scope.filterOptions.framework.id,
                     }).then((newSeriesList) => {
-                        $scope.platformList = _.union(
-                            $scope.platformList,
-                            [...new Set(_.map(newSeriesList, 'platform'))]);
-                        $scope.testList = _.union(
-                            $scope.testList,
-                            [...new Set(_.map(newSeriesList, 'name'))]);
+                        $scope.platformList = [...new Set([
+                            ...$scope.platformList,
+                            ...[...new Set(_.map(newSeriesList, 'platform'))]])];
```

Comment:
The double spread isn't needed here.  You can just do:

    ...new Set(_.map(newSeriesList, 'platform'))])];

Reply:
So you can simplify the other 3 double-spreads as well.